File: backend/app/core/api/v1/projects.py
"""Projects CRUD API."""

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import select, func
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.models.project import Project, ProjectBudget
from app.core.schemas.project import ProjectCreate, ProjectUpdate, ProjectResponse, ProjectListResponse

router = APIRouter(prefix="/projects", tags=["projects"])


@router.post("", response_model=ProjectResponse, status_code=201)
async def create_project(body: ProjectCreate, db: AsyncSession = Depends(get_db)):
    project = Project(
        title=body.title,
        genre=body.genre,
        format=body.format,
        description=body.description,
    )
    db.add(project)
    await db.flush()
    # Auto-create budget
    budget = ProjectBudget(project_id=project.id)
    db.add(budget)
    await db.flush()
    return project


@router.get("", response_model=ProjectListResponse)
async def list_projects(db: AsyncSession = Depends(get_db)):
    result = await db.execute(select(Project).order_by(Project.updated_at.desc()))
    projects = result.scalars().all()
    return ProjectListResponse(items=projects, total=len(projects))


@router.get("/{project_id}", response_model=ProjectResponse)
async def get_project(project_id: str, db: AsyncSession = Depends(get_db)):
    result = await db.execute(select(Project).where(Project.id == project_id))
    project = result.scalar_one_or_none()
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    return project


@router.delete("/{project_id}", status_code=204)
async def delete_project(project_id: str, db: AsyncSession = Depends(get_db)):
    result = await db.execute(select(Project).where(Project.id == project_id))
    project = result.scalar_one_or_none()
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    await db.delete(project)

File: frontend/src/lib/api/projects.ts
import { apiFetch } from './client'
import { Project, ProjectDetail, ProjectCreateRequest } from '@/types/project'
import { PaginatedResponse } from '@/types/api'

export async function createProject(request: ProjectCreateRequest): Promise<Project> {
  return apiFetch<Project>('/projects', {
    method: 'POST',
    body: JSON.stringify(request),
  })
}

export async function listProjects(): Promise<PaginatedResponse<Project>> {
  return apiFetch<PaginatedResponse<Project>>('/projects')
}

export async function getProject(projectId: string): Promise<ProjectDetail> {
  return apiFetch<ProjectDetail>(`/projects/${projectId}`)
}

export async function deleteProject(projectId: string): Promise<void> {
  return apiFetch<void>(`/projects/${projectId}`, { method: 'DELETE' })
}
